### DISCUSSION TITLE
Does helmfile support deploy chart has Library charts dependency

### DISCUSSION THREAD

**Discussion by gudeyang1** [helmfile/helmfile#545] [Questions]
Hi,  i'm new to helmfie, my chart are using helm [Library chart](https://helm.sh/docs/topics/library_charts/#helm) as dependency.

Normally, the deploy commands are 
```bash
helm pull myrepo/mychart  --untar
cd mychart
helm depdency update

helm upgrade --install
```
and the `Chart.yaml`
```bash

apiVersion: v2
name: baishan-api
description: A Helm chart for Kubernetes
version: 2.0.0
appVersion: 1.16.0
dependencies:
- name: common
  version: ">0.0.6"
  repository: "myrepo/common"
  import-values:
    - child: default
      parent: .
```
the `common` chart is a Library chart and can't  be deployed  independently.

i'm try to deploy the chart with helmfile `dependencies` block, but  helmfile treat the common chart as an independent chart and return error.
```bash

releases:
- name: baishan-api
  namespace: kip
  chart: baishan-api/baishan-api
  dependencies:
  - chart: common/common
    version: 0.0.32
```

![image](https://user-images.githubusercontent.com/13827211/204239966-135b80cc-731f-4e49-aac8-0384764b706b.png)

Is there a way to  fix the problem?  thanks

**Comment by yxxhero** [comment]
@gudeyang1 I saw your business chart that has a dependencies field. So you can try to remove dependencies in helmfile.

**Reply by gudeyang1** [reply]
thanks for the reply, i tried to remove the dependencies in helmfile but still got error.
```bash
ARGS:
  0: helm (4 bytes)
  1: diff (4 bytes)
  2: upgrade (7 bytes)
  3: --allow-unreleased (18 bytes)
  4: baishan-api (11 bytes)
  5: baishan-api/baishan-api (23 bytes)
  6: --namespace (11 bytes)
  7: kip (3 bytes)
  8: --detailed-exitcode (19 bytes)
  9: --color (7 bytes)
  10: --reset-values (14 bytes)

ERROR:
  exit status 1

EXIT STATUS
  1

STDERR:
  Error: Failed to render chart: exit status 1: Error: found in Chart.yaml, but missing in charts/ directory: common
  Error: plugin "diff" exited with error

COMBINED OUTPUT:
  ********************
  	Release was not present in Helm.  Diff will show entire contents as new.
  ********************
  Error: Failed to render chart: exit status 1: Error: found in Chart.yaml, but missing in charts/ directory: common
  Error: plugin "diff" exited with error

```
there are dependencies block in `Chart.yaml`,  it will fail if  no `common chart` found in charts directory.
is there some way to run `helm depdency update` before `helmfile apply`

**Reply by yxxhero** [reply]
@gudeyang1 please give a whole example to test. Thanks very much.

**Reply by yxxhero** [reply]
and show debug log by adding the --debug flag.
